### ISSUE TITLE
Main navigation menu

### ISSUE THREAD

**Issue by straight-shoota** [instedd/cdx#1550]
One of @jkicillof's  latest designs (https://projects.invisionapp.com/d/main#/console/4103589/465275288/comments) shows a main navigation menu with very few elements.
This is actually independent from the discussions about the transfer workflow and scanner interaction, so it makes sense to discuss this separately.

![grafik](https://user-images.githubusercontent.com/466378/159014323-c49d61d1-4b52-439a-ad96-c7756fd4edce.png)

I like it. I've been annoyed by the navigation menu being so big and hard to actually navigate.
After some time you're used to remember where the items are that you typically use.
But for new users it's overwhelming. And it's not great for experienced users either.

Joni's approach is to have a nested menu for samples and related things (batches, transfers).
It moves elements that are not relevant in the current context, out of sight. That makes them harder to reach, but you
typically don't need to reach them anyway from the sample page.

I had been thinking about another idea, though. The pages in the main navigation menu have very different usage characteristics.
For example, if you're working with samples, you'll spent most of the time on the samples page (and related).
In contrast, configuring a site or device is pretty rare. Many users may never have to go there after initial setup.

There is a clear difference in the frequency of these pages. Navigation should make frequently used things easily accessible,
and rarely used things can be further away.
I think the split goes right in the middle, where the first half are frequently used pages and the last half are rarely used.
Of course, this also depends on the kind of work you do. IIRC, UCSD has no patient interaction, so they probably don't use the patients page at all [^1].

In any case, I believe the last half of pages (sites, devices, logs, users, settings) are probably not frequently used.
They don't need to be easily reachable there when they're only used sporadically.
They are actually all kinds of settings. So I'd suggest to move them away from the main navigation menu and summarize all of them on the *Settings* page.
The link to that page could also be separated from the main entries, and moved to the right, next to the account menu.
It could also become a cog icon instead of a text label.
Then in the top right corner you have two menu items, one for account configuration and the other for context configuration. That fits together very nicely.

![grafik](https://user-images.githubusercontent.com/466378/159015362-01749dc1-0af4-45f4-bb90-96963136ae13.png)

As a result, we get a clean, small main navigation menu with all the important links easily accessible and less useful clutter tucked away, but still reachable.

[^1]: A further improvement could be to add configuration for hiding pages when they're not used.
  If an institution like UCSD doesn't use patients data, they could easily hide that button from the nav. They don't need to go there anyways.

**Comment by jkicillof** [comment]
I think there are different top navigations depending on the user type (institution, health organization, manufacturer). Maybe we could do the same for some kind of users that will only work with specific sections.

**Comment by straight-shoota** [comment]
@jkicillof @diegoliberman WDYT about moving *Sites*, *Devices*, *Logs*, and *Users* under *Settings*? This is a very easy UI change. It doesn't require any configuration. And it immediately clears up the navigation from less frequently used items.

We'd simply integrate those pages into the *Settings* landing page. I'd also change the *Settings* menu item to a cog icon.

Doing this we should also consider associating *Logs* with *Devices*, because it seems to be specifically about *Device logs*.

**Comment by diegoliberman** [comment]
I totally agree.  Go ahead in whatever gap or break you need from the current priorities.  Let Johny know so that he adapts the designs.  

Regarding logs, perhaps for now we can start just by renaming it to Device logs.

**Comment by jkicillof** [comment]
The screen you mentioned was an experiment I did before going with the samples menu. I agree with hiding those sections under settings. I'm not sure about the cog icon because those settings are not user settings and it might be confusing.

**Comment by jkicillof** [comment]
Updating all the screens won't be cheap, I could update easily all the ones that I did in sketch (samples, batche, boxes, transfers, etc).

**Comment by straight-shoota** [comment]
> Updating all the screens won't be cheap

IMO that might not really be necessary. This change doesn't really affect the main content of the screens, so I'd be fine with leaving the existing designs with the old nav.

We'll need an update for the settings page, though. To include the added items. And the current implementation diverts quite a bit from the design (https://projects.invisionapp.com/d/main?origin=v7#/console/4103589/119909353/preview?scrollOffset=14351).
I'll adjust that in a separate PR.

My idea for the settings page would be to group related entries, maybe even with group labels.
* Users/Rights: Roles, Policies, Users, Alerts (?)
* Devices: Devices, Device Logs, Device Models
* Other: Sites

**Comment by straight-shoota** [comment]
> I'm not sure about the cog icon because those settings are not user settings and it might be confusing.

IMO it's a common pattern to have two buttons, one for user settings/actions (indicated by the user icon) and one for app settings (indicated by the cog).
But I'm happy to implement any alternative suggestions =)

**Comment by jkicillof** [comment]
Great, I'll update the nav menu today EOD or tomorrow and I'll share the screen links

**Comment by jkicillof** [comment]
> IMO it's a common pattern to have two buttons, one for user settings/actions (indicated by the user icon) and one for app settings (indicated by the cog).

Yes, but in this case the settings are not user setting but organization settings. My concern is that this could get confused.

**Comment by straight-shoota** [comment]
Hm, I actually like the settings page. The action cards have contextual information and help texts about what the respective setting is about. IMO that's quite valuable and changing navigation to a plain dropdown menu would degrade the UX.
These settings pages are not used very often, so it's not that important to have them easily available in the main navigation. If a user wants to navigate there, it's not a big deal if they have to go to the settings overview page first. But because they're visited less often, it's also more important to inform the user what they are about.

A settings page is a good place to guide users on the available features in a better way than a list in a dropdown menu.
(Maybe I'm a bit driven by [this article](https://linear.app/blog/settings-are-not-a-design-failure) I recently read. It quite resonated with me the observation that settings pages are *"an excellent way to showcase our product and educate users about all its functionalities."*)

**Comment by jkicillof** [comment]
Yes, I think both options could work. I'll have to work on the missing illustrations. I'll try to get them asap.

**Comment by jkicillof** [comment]
I updated the [settings screen](https://projects.invisionapp.com/d/main#/console/4103589/119909353/preview). You should be able to donwload each icon.
